src/main/java/com/dropbox/core/DbxRequestUtil.java

```diff
@@ -278,44 +278,63 @@ public static String parseErrorBody(String requestId, int statusCode, byte[] bod
+                break;
             case 401:
-                return new InvalidAccessTokenException(requestId, message);
+                networkError = new InvalidAccessTokenException(requestId, message);
+                break;
             case 429:
                 try {
                     int backoffSecs = Integer.parseInt(getFirstHeader(response, "Retry-After"));
-                    return new RateLimitException(requestId, message, backoffSecs, TimeUnit.SECONDS);
+                    networkError = new RateLimitException(requestId, message, backoffSecs, TimeUnit.SECONDS);
+                    break;
                 } catch (NumberFormatException ex) {
-                    return new BadResponseException(requestId, "Invalid value for HTTP header: \"Retry-After\"");
+                    networkError = new BadResponseException(requestId, "Invalid value for HTTP header: \"Retry-After\"");
+                    break;
                 }
             case 500:
-                return new ServerException(requestId, message);
+                networkError = new ServerException(requestId, message);
+                break;
             case 503:
                 // API v1 may include Retry-After in 503 responses, v2 does not
                 String retryAfter = getFirstHeaderMaybe(response, "Retry-After");
                 try {
                     if (retryAfter != null && !retryAfter.trim().isEmpty()) {
                         int backoffSecs = Integer.parseInt(retryAfter);
-                        return new RetryException(requestId, message, backoffSecs, TimeUnit.SECONDS);
+                        networkError = new RetryException(requestId, message, backoffSecs, TimeUnit.SECONDS);
+                        break;
                     } else {
-                        return new RetryException(requestId, message);
+                        networkError = new RetryException(requestId, message);
+                        break;
                     }
                 } catch (NumberFormatException ex) {
-                    return new BadResponseException(requestId, "Invalid value for HTTP header: \"Retry-After\"");
+                    networkError = new BadResponseException(requestId, "Invalid value for HTTP header: \"Retry-After\"");
+                    break;
                 }
             default:
-                return new BadResponseCodeException(
+                networkError = new BadResponseCodeException(
                     requestId,
                     "unexpected HTTP status code: " + response.getStatusCode() + ": " + message,
                     response.getStatusCode()
                 );
         }
+
+        DbxGlobalCallbackFactory factory = DbxRequestUtil.sharedCallbackFactory;
+        if (factory != null) {
+            DbxNetworkErrorCallback callback = factory.createNetworkErrorCallback();
+            callback.setNetworkError(networkError);
```

Comment:
Does `DbxNetworkErrorCallback` need to be a `Runnable`? Instead of having `setNetworkError()` + `run()`, would it be simpler to just have one method like `onNetworkError(T error)`?